test/groups/test_general_linear_group.jl

```diff
@@ -0,0 +1,72 @@
+using LieGroups, Random, Test
+
+using ManifoldsBase: ℂ
+
+s = joinpath(@__DIR__, "..", "LieGroupsTestSuite.jl")
+!(s in LOAD_PATH) && (push!(LOAD_PATH, s))
+using LieGroupsTestSuite
+
+@testset "GL(2)" begin
+    G = GeneralLinearGroup(2)
+    g1, g2, g3 = [2.0 0.0; 0.0 1.0], [1.0 0.5; 0.5 1.0], [1.0 2.0; 3.0 4.0]
+    X1, X2, X3 = [-1.0 0.0; 0.0 0.0], [0.0 0.5; 0.5 0.0], [1.0 2.0; 3.0 4.0]
+    properties = Dict(
+        :Name => "The general linear group",
+        :Points => [g1, g2, g3],
+        :Vectors => [X1, X2, X3],
+        :Rng => Random.MersenneTwister(),
+        :Functions => [
+            compose,
+            conjugate,
+            diff_conjugate,
+            diff_inv,
+            diff_left_compose,
+            diff_right_compose,
+            exp,
+            # hat, # requires a fix in Manifolds.jl to have an ONB on invertible matrices
+            inv,
+            inv_left_compose,
+            inv_right_compose,
+            is_identity,
+            lie_bracket,
+            log,
+            #rand, # requires a fix in Manifolds.jl to have rand on invertible matrices
+            show,
+            #vee, # requires a fix in Manifolds.jl to have an ONB on invertible matrices
+        ],
+    )
+    expectations = Dict(
+        :repr => "GeneralLinearGroup(2; field=ℝ)", :lie_bracket => X1 * X2 - X2 * X1
+    )
+    test_lie_group(G, properties, expectations)
+
+    @test_broken is_point(G, Identity(G); error=:error)
```

Comment:
One last thing that used to work in Manifolds.jl but doesn't here.

Reply:
Uff. I tried to fix this, because I do already provide the right `check_point` functions a bit, so I am not sure I have to maybe reimplement the whole dispatch path? Something along that is erroring :/

Reply:
The usual thing when one tries to say: I am out of ideas: I saw the right idea: `check_size` was missing, but that is just nothing because identity is of right size anyways; check points were adapted for the positive (same op) and the negative case (different op).

Reply:
Or maybe not...

Uff, this is complicated. It worked for short but now it even does not compile. I'll take a break.